cdap-app-fabric/src/main/java/io/cdap/cdap/app/DefaultAppConfigurer.java

```diff
@@ -116,6 +119,7 @@ public DefaultAppConfigurer(Id.Namespace namespace, Id.Artifact artifactId, Appl
     this.appMetadata = new HashMap<>();
     this.triggerFactory = new DefaultTriggerFactory(namespace.toEntityId());
     this.runtimeConfigurer = runtimeConfigurer;
+    this.featureFlagsProvider = this;
```

Comment:
Why?? This seems recursive. If the provider is given to the parent class, we shouldn't need this field in this class.

Reply:
You can have a `protected final FeatureFlagsProvider getFeatureFlagsProvider()` method in the `AbstractConfigurer` to expose it.